### ISSUE TITLE
Huge box sizes in GRO-files

### ISSUE THREAD

**Issue by PicoCentauri** [MDAnalysis/mdanalysis#2371]
I currently facing the problem of HUGE box sizes in my GRO files. Due to the [current implementation](https://github.com/MDAnalysis/mdanalysis/blob/9372d093acebb78bc5b1a2b4dba815a4e3281b04/package/MDAnalysis/coordinates/GRO.py#L203) they can not be parsed at all. To fix this one can just replace

```python
unitcell = np.float32(line.split())
```

by 

```python
split = re.split("(\......)", line)
unitcell = np.float32([i+j for i, j in zip(split[::2], split[1::2])])
```
since the box dimensions always contain 5 decimals places.

### Expected behaviour

Consider the following GRO of a cubic box with a box length of 40000 nm.

```
huge cubic box
    1
    1SOL      H    1   0.000   0.000   0.000
40000.0000040000.0000040000.00000
```

This should give a dimension array of `[4.e+05 4.e+05 4.e+05 9.e+01 9.e+01 9.e+01]`

### Actual behaviour

ValueError during parsing
```python
ValueError: could not convert string to float: '40000.0000040000.0000040000.00000'
```

### Code to reproduce the behaviour

``` python
import MDAnalysis as mda

u = mda.Universe("cubic.gro")
```

### Currently version of MDAnalysis:
0.20.2-dev0

**Comment by orbeckst** [comment]
Can you dig up a current file format spec for GRO? The link on https://www.mdanalysis.org/docs/documentation_pages/coordinates/GRO.html to http://manual.gromacs.org/current/online/gro.html is broken.

I want to know if the file format specifies fixed width or space separated for the box. This determines if this issue is a bug or a feature request.

**Comment by PicoCentauri** [comment]
http://manual.gromacs.org/current/reference-manual/file-formats.html?#gro

The docu states: "box vectors (free format, space separated reals)". But GROMACS itself does not care for this and produces these weird files...

**Comment by zemanj** [comment]
Looks like a gromacs bug to me...

**Comment by PicoCentauri** [comment]
Just for our interest: The [current implementation](https://github.com/MDAnalysis/mdanalysis/blob/9372d093acebb78bc5b1a2b4dba815a4e3281b04/package/MDAnalysis/coordinates/GRO.py#L292) of the `GROWriter` also produces similiar files without a space if the box is large enough.

**Comment by richardjgowers** [comment]
No harm in putting the regex version in an except branch (after trying
split)

**Comment by zemanj** [comment]
The format strings in `GROWriter.fmt` should be fixed to enforce spaces between box dimensions

**Comment by orbeckst** [comment]
The work-around is fugly but if Gromacs decides to do that... I would still spit out a warning saying that the GRO format does not implement the file format standard and that the unit cell might be wrong.

+1 to fixing our writer!

**Comment by PicoCentauri** [comment]
> The format strings in `GROWriter.fmt` should be fixed to enforce spaces between box dimensions

Yes, this I can do.

> The work-around is fugly but if Gromacs decides to do that...

I also raised an issue on the [Gromacs Bugtracker](https://redmine.gromacs.org/issues/3176). Lets wait for a while an see what they decide. If they don't want to change their output format I would like to put the regexp version into the parser even if its ugly...

**Comment by ezavod** [comment]
May I add a few notes:
First, even Gromacs is unable to read back their own malformed files correctly, so +1 for having a warning that something is wrong.
This could be shown based on @PicoCentauri 's example (box of `tmp.gro` is wrong):
```
gmx solvate -box 1 1 1
gmx editconf -f out.gro -box 1111 2222 3333 -o bug.gro
gmx editconf -f bug.gro -o tmp.gro   // box is '1111.00000   0.00000   0.00000'
```
Second, may I suggest using `re.findall()` to avoid confusing zipping and using a more verbose regex `(\d+\.\d{5})`:
```python
line = "100000.0000010000.000001000.00000"
finds = re.findall(r"(\d+\.\d{5})", line)
assert(len(finds) == 3 or len(finds) == 9)
unitcell = np.float32(finds)
print(unitcell)
```

**Comment by mabraham** [comment]
Fix proposed for upcoming GROMACS 2019.5: https://gerrit.gromacs.org/c/gromacs/+/14117
